### DISCUSSION TITLE
XIAO_MG24 TxPower Setting

### DISCUSSION THREAD

**Discussion by msfujino** [SiliconLabs/arduino#109] [Q&A]
I am experimenting with the newly supported XIAO_MG24.

I used the function sl_status_t sl_bt_advertiser_set_tx_power() to set the transmit power from -20 to 20 dBm. The graph below plots the RSSI on the receive side.

1. Settings above 10 dBm appear to be limited to the global value of 10 dBm. I used the function sl_bt_system_set_tx_power() to set the global value to 20dBm, but it does not work.
2. I do not know the reason for the strange change around -10dBm.

Does anyone have any information on this?

![MG24_TxPower](https://github.com/user-attachments/assets/9b4717fc-24d9-4cc6-b879-0b6c2ba59457)

**Comment by ggtimtom** [comment]
The discription of the function:

Limit the maximum advertising TX power on an advertising set. If the value goes over the global value that was set using the @ref sl_bt_system_set_tx_power command, the global value will be the maximum limit. The maximum TX power of legacy advertising is further constrained to be less than +10 dBm. Extended advertising TX power can be +10 dBm and over if Adaptive Frequency Hopping is enabled. This setting has no effect on periodic advertising.
This setting will take effect next time the legacy or extended advertising is enabled.
By default, maximum advertising TX power is limited by the global value.

**Reply by msfujino** [reply]
Thanks for the reply.
Of course I have already read this document.

Regarding the function **sl_bt_system_set_tx_power()**, I found the following description in **sl_bt_api.h**: L1278-L1317
“By default, the minimum radiated TX power level is configured to -3 dBm and the maximum radiated TX power level to 8 dBm.”
